src/deploy/functions/release.ts

```diff
@@ -2,477 +2,120 @@
+          helper.printSuccess(fn.name, "upsert schedule");
+        })
+        .catch((err) => {
+          errorHandler.record("error", fn.name, "upsert schedule", err.message || "");
         });
-      });
-    } else {
-      if (deployments.length !== 0) {
-        utils.logBullet(clc.bold.cyan("functions: ") + "continuing with other deployments.");
-      }
     }
   }
-  // filter out functions that are excluded via --only and --except flags
-  const functionsInDeploy = functionsInfo.filter((trigger: CloudFunctionTrigger) => {
-    return functionFilterGroups.length > 0 ? _.includes(deleteReleaseNames, trigger.name) : true;
-  });
-  await createOrUpdateSchedulesAndTopics(
-    context.projectId,
-    functionsInDeploy,
-    existingScheduledFunctions,
-    appEngineLocation
-  );
-  const allOps = await utils.promiseAllSettled(
-    _.map(deployments, async (op) => {
-      const res = await op.retryFunction();
-      return _.merge(op, res);
-    })
-  );
-  const failedCalls = _.chain(allOps).filter({ state: "rejected" }).map("reason").value();
-  const successfulCalls = _.chain(allOps).filter({ state: "fulfilled" }).map("value").value();
-  failedDeployments = failedCalls.map((error) => _.get(error, "context.function", ""));
-
-  await fetchTriggerUrls(projectId, successfulCalls, sourceUrl);
-
-  await helper.pollDeploys(successfulCalls, printSuccess, printFail, printTooManyOps, projectId);
-  if (deployments.length > 0) {
-    track("Functions Deploy (Result)", "failure", failedDeployments.length);
-    track("Functions Deploy (Result)", "success", deployments.length - failedDeployments.length);
+  for (const fnName of fullDeployment.schedulesToDelete) {
+    schedulerQueue
+      .run(tasks.deleteScheduleTask(fnName, appEngineLocation))
+      .then(() => {
+        helper.printSuccess(fnName, "delete schedule");
+      })
+      .catch((err) => {
+        errorHandler.record("error", fnName, "delete schedule", err.message || "");
+      });
   }
 
-  if (failedDeployments.length > 0) {
-    logger.info("\n\nFunctions deploy had errors with the following functions:");
-    const sortedFailedDeployments = failedDeployments.sort();
-    for (const failedDep of sortedFailedDeployments) {
-      logger.info(`\t${failedDep}`);
-    }
-    logger.info("\n\nTo try redeploying those functions, run:");
-    logger.info(
-      "    " +
-        clc.bold("firebase deploy --only ") +
-        clc.bold('"') +
-        clc.bold(
-          sortedFailedDeployments.map((name) => `functions:${name.replace(/-/g, ".")}`).join(",")
-        ) +
-        clc.bold('"')
-    );
-    logger.info("\n\nTo continue deploying other features (such as database), run:");
-    logger.info("    " + clc.bold("firebase deploy --except functions"));
-    throw new FirebaseError("Functions did not deploy properly.");
-  }
+  // Once everything has been added to queues, starting processing.
+  // Note: We need to set up these wait before calling process and close.
```

Comment:
Figuring this out drove me up a wall for a day and a half :(. It turns out that our implementation of throttler.wait() returns a promise that never resolves if you call it after .process() and .close(). I'd like to fix this at some point, but that code is used in a bunch of other places that I don't understand well enough to test effectively.